evaluate.py

```diff
@@ -55,46 +56,51 @@ def generate(args, g_ema, device, mean_joints, std_joints, entity):
         index = range(len(generated_motion))
 
     generated_motion_np, _ = get_gen_mot_np(args, generated_motion, mean_joints, std_joints)
-    generated_motions = np.concatenate(generated_motion_np, axis=0)
 
-    if entity.str() == 'Joint':
-        return generated_motions
+    motion_data_raw = np.load(args.path, allow_pickle=True)
+    offsets = np.concatenate([motion_data_raw[0]['offset_root'][np.newaxis, :] * 0, motion_data_raw[0]['offsets_no_root']])
+    if args.dataset == 'mixamo':
+        offsets /= 100  # not needed in humanact  TODO: Why do we need that?
 
-    _, _, _, edge_rot_dict_general = motion_from_raw(args, np.load(args.path, allow_pickle=True))
-    generated_motions = convert_motions_to_location(args, generated_motion_np, edge_rot_dict_general)
-    return generated_motions
+    static = StaticData(parents=motion_data_raw[0]['parents_with_root'],
+                        offsets=offsets,
+                        names=motion_data_raw[0]['names_with_root'],
+                        character_name=args.character,
+                        n_channels=4,
+                        enable_global_position=args.glob_pos,
+                        enable_foot_contact=args.foot,
+                        rotation_representation=args.rotation_repr)
 
+    dynamic_data = torch.stack([torch.from_numpy(motion[0]) for motion in generated_motion_np]).transpose(1, 2)
+    dynamics = DynamicData(dynamic_data, static, use_velocity=args.use_velocity)
+    dynamics = dynamics.un_normalise(mean_joints.transpose(0, 2, 1, 3), std_joints.transpose(0, 2, 1, 3))
+
+    generated_motions = convert_motions_to_location(dynamics, args.dataset)
+
+    return generated_motions
 
-def convert_motions_to_location(args, generated_motion_np, edge_rot_dict_general):
-    edge_rot_dict_general['std_tensor'] = edge_rot_dict_general['std_tensor'].cpu()
-    edge_rot_dict_general['mean_tensor'] = edge_rot_dict_general['mean_tensor'].cpu()
-    if args.dataset == 'mixamo':
-        edge_rot_dict_general['offsets_no_root'] /= 100 ## not needed in humanact
 
+def convert_motions_to_location(dynamics, dataset_type):
     generated_motions = []
 
```

Comment:
Throughout the whole code: if not specifically dealing with static/dynamic attributes, don't call a motion 'dynamics'. Call it 'motion', or 'dynamic_motion', or similar.
Do it for all your changes

Reply:
Here, for example, it is important for the user to know that the argument is a motion and that it's a generated one. So the name 'generated_motion' makes sense. When the user sees the name 'dynamic' he thinks: what is this argument? a dynamic what?